plotly/tests/test_core/test_plotly/test_plot.py

```diff
@@ -66,3 +70,187 @@ def test_plot_empty_data(self):
+
+        plot_option_logic = py._plot_option_logic(kwargs)
+
+        expected_plot_option_logic = {'filename': 'test',
+                                      'auto_open': True,
+                                      'fileopt': 'overwrite',
+                                      'validate': True,
+                                      'world_readable': False,
+                                      'sharing': None}
+        self.assertEqual(plot_option_logic, expected_plot_option_logic)
+
+    def test_plot_option_logic_only_sharing_given(self):
+
+        # If world_readable is not given and sharing ='private',
+        # world_readable should be set to False
+
+        kwargs = {'filename': 'test',
+                  'auto_open': True,
+                  'fileopt': 'overwrite',
+                  'validate': True,
+                  'sharing': 'private'}
+
+        plot_option_logic = py._plot_option_logic(kwargs)
+
+        expected_plot_option_logic = {'filename': 'test',
+                                      'auto_open': True,
+                                      'fileopt': 'overwrite',
+                                      'validate': True,
+                                      'world_readable': False,
+                                      'sharing': 'private'}
+        self.assertEqual(plot_option_logic, expected_plot_option_logic)
+
+    # @skip('added to see if everything else passes on circle')
+    def test_plot_url_given_sharing_key(self):
+
+        # Give share_key is requested, the retun url should contain
+        # the share_key
+
+        data = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
+        validate = True
+        fig = tls.return_figure_from_figure_or_data(data, validate)
+        kwargs = {'filename': 'is_share_key_included',
+                  'fileopt': 'overwrite',
+                  'world_readable': False,
+                  'sharing': 'secret'}
+        response = py._send_to_plotly(fig, **kwargs)
+        plot_url = response['url']
+
+        self.assertTrue('share_key=' in plot_url)
+
+    def test_plot_url_response_given_sharing_key(self):
+
+        # Give share_key is requested, the get request of url should
+        # be 200
+
+        data = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
+        validate = True
+        fig = tls.return_figure_from_figure_or_data(data, validate)
+        kwargs = {'filename': 'is_share_key_included',
+                  'fileopt': 'overwrite',
+                  'world_readable': False,
+                  'sharing': 'secret'}
+
+        self.assertTrue(200 ==
+                        requests.get(
+                            py._send_to_plotly(fig, **kwargs)['url'])
+                        .status_code)
```

Comment:
:cow2: This makes debugging the test pretty hard. It'd be better to write it like this:

```
# explain why we need to use _send_to_plotly instead of .plot or something?
plot_url = py._send_to_plotly(fig, **kwargs)['url']
response = requests.get(plot_url)
self.assertEqual(response.status_code, 200)
```

Same amount of lines and it will be much clearer to understand and debug.